Sources/SwiftSyntax/SyntaxNodes.swift.gyb.template

```diff
@@ -113,35 +113,54 @@ public struct ${node.name}: ${base_type}Protocol, SyntaxHashable {
     self._syntaxNode = Syntax(data)
   }
 
-  public init(
-%     for (index, child) in enumerate(node.children):
-%       comma = ',' if index != len(node.children) - 1 else ''
-%       param_type = child.name if child.node_choices else child.type_name
-%       if child.is_optional:
-%         param_type = param_type + "?"
-%       if child.is_unexpected_nodes():
-    _ ${child.swift_name}: ${param_type} = nil${comma}
-%       else:
-    ${child.swift_name}: ${param_type}${comma}
-%       end
-%     end
-  ) {
+  ${node.generate_initializer_decl(True, "  ")} {
```

Comment:
I think this is a lot clearer with argument labels (same below).

```suggestion
  ${node.generate_initializer_decl(include_optional_base=True, current_indentation="  ")} {
```

Reply:
Heh, yeah fair enough. Thanks